File: app/src/main/java/com/agrogames/islandsofwar/render/impl/ObjectRenderer.java
package com.agrogames.islandsofwar.render.impl;

import android.util.Pair;

import com.agrogames.islandsofwar.engine.abs.common.Point;
import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject;
import com.agrogames.islandsofwar.graphics.abs.TextureDrawer;

public class ObjectRenderer {
    private final TextureDrawer drawer;

    public ObjectRenderer(TextureDrawer drawer) {
        this.drawer = drawer;
    }

    public float render(RenderableObject object){
        return simpleRender(object, TextureMapper.join(object.getTexture()));
    }

    public float render(RenderableObject object, String state){
        return simpleRender(object, TextureMapper.join(object.getTexture(), state));
    }

    private float simpleRender(RenderableObject object, String texture){
        Point location = object.getLocation();
        Pair<Float, Float> size = drawer.drawTexture(location.x, location.y, texture, object.getRotation());
        return size.first > size.second ? size.first : size.second;
    }
}

File: app/src/main/java/com/agrogames/islandsofwar/render/impl/ObjectRenderer.kt
package com.agrogames.islandsofwar.render.impl

import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject
import com.agrogames.islandsofwar.graphics.impl.drawtexture.TextureDrawer

class ObjectRenderer(private val drawer: TextureDrawer) {
    fun render(`object`: RenderableObject): Float {
        return simpleRender(`object`, TextureMapper.join(`object`.texture))
    }

    fun render(`object`: RenderableObject, state: String): Float {
        return simpleRender(`object`, TextureMapper.join(`object`.texture, state))
    }

    private fun simpleRender(`object`: RenderableObject, texture: String): Float {
        val location = `object`.location
        val size = drawer.drawTexture(location.x, location.y, texture, `object`.rotation)
        return if (size.first!! > size.second!!) size.first!! else size.second!!
    }
}